mern/client/src/Contexts/sessionProvider.js

```diff
@@ -6,10 +6,18 @@ export default function SessionProvider({ children }) {
   // nav state to show or hide session id/save/delete buttons
   const [navState, setNavState] = React.useState(false);
   const [transcriptID, setTranscriptID] = React.useState();
```

Comment:
This file context is getting quite big in terms number of states. Are all of them necessary to be in sessionProvider? Also what is the difference between navState and transcriptID

Reply:
Yeah maybe I can move the transcriptID to a different context file? I felt that it was relevant to the session so I just kept it there for now. 

The difference between navState and transcriptID is that the navState is either true or false (when true, it shows the save and delete buttons on the top, when false, it hides it), and for the transcriptID, I'm using to store the string of the transcriptID the creator uploads so we can pass it into functions such as deleting the transcript from the DB.

Reply:
But can't we just check if transcriptID is defined to see if the session has been entered?

Reply:
Or is it better for the single responsibility principle to split each of those things into a different state?

Reply:
Hmmmmmmm now I'm thinking, the only reason you would want the navState to not show is when you haven't started a session meaning no transcript has been uploaded.......... I think you're right, we can check if transcriptID is defined for the navState. Unless recover session does something different? Does recovering the session set the transcriptID?